src/util/path.js

```diff
@@ -465,6 +465,28 @@
     return tmpLen;
   }
 
+  // add explanation in docs
+  function findPercentageForDistance(segInfo, distance) {
+    var perc = 0, tmpLen = 0, iterator = segInfo.iterator, tempP = { x: segInfo.x, y: segInfo.y },
+        p, nextLen, nextStep = 0.01;
+    while (tmpLen < distance && perc <= 1 && nextStep > 0.0000001) {
+      p = iterator(perc);
+      nextLen = calcLineLength(tempP.x, tempP.y, p.x, p.y);
+      // compare tmpLen each cycle with distance, decide next perc to test.
+      if ((nextLen + tmpLen) > distance) {
+        // we discard this step and we make smaller steps.
+        nextStep /= 2;
+        perc -= nextStep;
+      }
+      else {
+        tempP = p;
+        perc += nextStep;
+        tmpLen += nextLen;
+      }
+    }
+    return p;
+  }
```

Comment:
steps are: 
```
0.01, 
0.005, 
0.0025. 
0.00125,
0.000625
0.0003125
0.00015625 // this is 1/64 of a 1%, i think maybe this is enough?
0.000078125
0.0000390625
0.00001953125
0.000009762625
```
Maybe i do not need to go down to 0.0000001

Reply:
Or maybe i do not need to start from 0 every time.
I can return the point and the percentage i calculated, and feed it back at the next char.

Reply:
Surely 0.0015625 is enough precision. I would think though that either way shouldn't cause too much of a performance hit since the step amount only gets reduced when within less than 1%. But yes, having it avoid duplicating calculations for each character in a segment would make sense.